supabase/models/updateRequestToReserve.ts

```diff
@@ -0,0 +1,39 @@
+import newClient from '../utils/newClient';
+
+export const updateRequestToReserve = async (
+  itemId: number,
+  userId: string
+) => {
+  try {
+    const supabase = newClient();
+
+    const { data: item, error: fetchError } = await supabase
+      .from('items')
+      .select('requestedToReserve')
+      .eq('id', itemId)
+      .single();
+
+    if (fetchError) {
+      throw new Error(fetchError.message);
+    }
+
+    const currentRequestedToReserve = item?.requestedToReserve || [];
+    if (currentRequestedToReserve.includes(userId)) {
+      alert('You have already requested this item.');
+    } else {
+      const updatedRequestedToReserve = [...currentRequestedToReserve, userId];
+      const { error: updateError } = await supabase
+        .from('items')
+        .update({ requestedToReserve: updatedRequestedToReserve })
+        .eq('id', itemId);
+
+      if (updateError) {
+        throw new Error(updateError.message);
+      }
+      alert('Request to reserve the item has been sent successfully.');
```

Comment:
This is working fine but it's really inconsistent with what we've been doing so far 🤔 maybe we can pass the button a setter for a raised state and update it here so that the parent component shows a "success" or "fail" message instead?

Reply:
Done

Fix reserve item functionality and display status messages
- Implemented state management for reservation requests using `useState` hook.
- Updated `handleReserve` function to handle errors and update status messages.
- Conditionally rendered status messages below the item details.